src/rpc/quorums.cpp

```diff
@@ -874,20 +875,37 @@ static UniValue verifychainlock(const JSONRPCRequest& request)
 
     const uint256 nBlockHash(ParseHashV(request.params[0], "blockHash"));
 
-    CBLSSignature sig;
-    if (!sig.SetHexStr(request.params[1].get_str())) {
-        throw JSONRPCError(RPC_INVALID_PARAMETER, "invalid signature format");
-    }
-
     int nBlockHeight;
+    CBlockIndex* pIndex{nullptr};
     if (request.params[2].isNull()) {
-        const CBlockIndex* pIndex = WITH_LOCK(cs_main, return g_chainman.m_blockman.LookupBlockIndex(nBlockHash));
+        pIndex = WITH_LOCK(cs_main, return g_chainman.m_blockman.LookupBlockIndex(nBlockHash));
         if (pIndex == nullptr) {
             throw JSONRPCError(RPC_INTERNAL_ERROR, "blockHash not found");
         }
         nBlockHeight = pIndex->nHeight;
     } else {
         nBlockHeight = ParseInt32V(request.params[2], "blockHeight");
+        LOCK(cs_main);
+        if (nBlockHeight < 0) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Block height out of range");
+        }
+        if (nBlockHeight <= ::ChainActive().Height()) {
+            pIndex = ::ChainActive()[nBlockHeight];
+        }
+    }
+
+    CBLSSignature sig;
+    if (pIndex) {
```

Comment:
if we reach here, pIndex will always be non-null due to L881 or L893 (only avoided by throwing)